### PR TITLE
add log analytics

### LINKED ISSUE #7: terraform - App Services
## Description
- App Service fails to pull repo
- ACR ID and Key are stored in clear text

### LINKED ISSUE #9: terraform - webv
- remove App Insights
- add Log Analytics
- name smoker HeName-webv-location to be consistent
- deploy smokers in eastus2, westeurope, southeastasia
- change command line to be consistent with quick start setup

### PR DESCRIPTION
**Pull request by bartr**
## Type of PR

- [ ] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

### Purpose of PR
- added log analytics to TF script
- create 4 smokers

Closes #7 
Closes #9

### CODE REVIEW

File: src/modules/webapp/main_alerts.tf [hunk 1]
```diff
@@ -1,25 +1,49 @@
 resource "random_uuid" "appsguid" {}
 resource "random_uuid" "webtestguid" {}
 
-
 resource "azurerm_monitor_action_group" "helium-action-group" {
-  name                = var.ACTION_GROUP_NAME
-  resource_group_name = var.APP_RG_NAME
-  short_name          = substr("${var.ACTION_GROUP_NAME}", 0, 12)
+  name                      = "${var.NAME}-acton-group"
```
Comment (wereinse) [inline src/modules/webapp/main_alerts.tf:5]: seems like the 12 character length restriction is gone - I tested creating an action group in the 20s.  Typo on line 5.

File: todo-list [hunk 2]
```diff
@@ -1,71 +1,33 @@
...
-- wait for acr import before creating acr web hook
-- wait for web app before creating acr hook?
-
-- should the tf state RG be created with the same TF script? A tf destroy will delete the state
-- how do the state files get copied to Azure? there is no storage account created
-
-- Should we add webv deployment to manual instructions for consistency?
+- need to address the design
+  - currently, everything is deployed as a single TF apply
+  - per our original design, cosmos and ACR have separate lifecyles from the app / webv
```
Comment (wereinse) [inline todo-list:5]: agreed - maybe there should be a terraform update script / tfvars

File: todo-list [hunk 3]
```diff
@@ -1,71 +1,33 @@
...
-- I had to run imdb-import manually to get webv to pass
-  - this now runs syncronously and blocks web
-- web app throws errors due to timing
-  - waits for imdb-import
-  - this blocks smoker
-- smoker throws errors due to timing
-  - waits for imdb-import and web deploy
+- Web Deploy
+  - ACR ci-cd webhook is setup - need to test
+  - docker container logging isn't enabled
```
Comment (wereinse) [inline todo-list:14]: The webv containers do have logging - but we can turn on in the docker pull statement also.
I can ping the webhook via the portal but I believe it's coming from docker.  The ACR is empty.

File: todo-list [hunk 4]
```diff
@@ -1,71 +1,33 @@
...
 
-- the image is pulled from docker hub
-  - should be loaded into ACR (done)
-  - will need an additional SP (done)
-  - will need ACR pull permissions on the SP (done)
+- there is an error in the app service log
+  - the first time the app starts, key vault isn't set and you get a usage message
+    - can we add timing to prevent?
```
Comment (wereinse) [inline todo-list:18]: Yes - we could add an output to src/web/main_kv.tf here resource "azurerm_key_vault_access_policy" "web_app".  Then create a depends_on for the app to wait on that.

File: todo-list [hunk 5]
```diff
@@ -1,71 +1,33 @@
...
-  - added He_Repo sed
-- fixed ACR import bug
-- added He_Name preface to tfstate-rg
-- standardized casing / naming in dashboard / alerts
+- we should be able to use an object for the dashboard variables so we only have to pass one variable instead of each individually
 
+- module readme files need to be updated and regenerated
 
-## Features
+- need to test dashboard and alerts to make sure they are the same between manual setup and terraform setup
```
Comment (wereinse) [inline todo-list:29]: Yes that would be a good task - they for sure are not.

File: src/modules/webapp/main_alerts.tf (hunk 1 above)
Comment (bartr) [reply src/modules/webapp/main_alerts.tf:5]: create-tf-vars throws an error if you create a name > 12

**Comment by bartr** [comment]
since Azure blobs are encrypted, there's no need to encrypt the tfstate file. That said, I didn't change anything except adding the depends on

**Pull request merged**